### PR TITLE
Master enhance spreadsheet pivot pro

### PR DESCRIPTION
**Pull request by pro-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js [hunk 1]
```diff
@@ -99,15 +99,6 @@ export class PivotCorePlugin extends OdooCorePlugin {
                 this._insertPivot(sheetId, position, id, spTable);
                 break;
             }
-            case "RE_INSERT_PIVOT": {
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:112]: :ok_hand:

**Review by LucasLefevre — COMMENTED** [review]
Some comments/suggestions (that can be left for later though)

File: addons/spreadsheet/static/src/@types/commands.d.ts [hunk 2]
```diff
@@ -52,17 +52,32 @@ declare module "@spreadsheet" {
...
     id: string;
     sheetId: string;
     col: number;
     row: number;
-    payload: {
-      type: "ODOO";
-      table: SPTableData;
-      definition: OdooPivotDefinition;
-    };
+    payload: InsertPivotPayload;
```
[the remark this answers is not in the document]
Comment (pro-odoo) [reply addons/spreadsheet/static/src/@types/commands.d.ts]: Nope, `ExtendedInsertPivotPayload` is used in `ExtendedInsertPivotCommand`. `InsertPivotCommand` is the one that should be moved to o-spreadsheet

File: addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js [hunk 3]
```diff
@@ -100,12 +81,10 @@ export class PivotCorePlugin extends OdooCorePlugin {
...
             case "INSERT_PIVOT": {
                 const { sheetId, col, row, id, payload } = cmd;
-                const { definition } = payload;
                 /** @type { { col: number, row: number } } */
                 const position = { col, row };
-                const { cols, rows, measures, rowTitle } = payload.table;
-                const table = new SpreadsheetPivotTable(cols, rows, measures, rowTitle);
-                this._addPivot(id, definition);
+                const table = pivotRegistry.get(payload.type).getTable(payload);
+                this._addPivot(id, payload);
```
[the remark this answers is not in the document]
Comment (pro-odoo) [reply addons/spreadsheet/static/src/pivot/plugins/pivot_core_plugin.js:89]: In this commit, it's done directly in the import/export method. Or I missed something ?
But in any case, as you said, it's done in the next commit

File: addons/spreadsheet/static/src/@types/commands.d.ts [hunk 4]
```diff
@@ -58,8 +58,11 @@ declare module "@spreadsheet" {
     sheetId: string;
     col: number;
     row: number;
-    table: SPTableData;
-    definition: OdooPivotDefinition;
+    payload: {
```
[the remark this answers is not in the document]
Comment (pro-odoo) [reply addons/spreadsheet/static/src/@types/commands.d.ts]: Makes a lot of sense

**Review by LucasLefevre — APPROVED** [review]
robodoo rebase-ff r+

**Comment by LucasLefevre** [comment]
robodoo r+

**Pull request closed without merging**